### PR TITLE
Update to use esptool-js

### PR DESCRIPTION
**Pull request by makermelissa**
This PR completely guts the ESPTool that did the programming and replaces the underlying code with esptool-js while still retaining the same user interface. This should be a 2.0.0 release, though since nothing is compiled anymore, that isn't strictly necessary.

I also took into account the passthrough option (which wasn't saving the setting anymore) when updating.

cc @dhalbert and @ladyada for review

### CODE REVIEW

**Review by dhalbert — CHANGES_REQUESTED** [review]
I think too many files were deleted, including `README.md`, `package.json`, etc.?

**Comment by ladyada** [comment]
ok thanks @tyeth can test it!

**Comment by makermelissa** [comment]
All typescript and related files were removed because the that functionality is now handled by esptool-js.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+## Used by
+
+- [Adafruit WipperSnapper](https://learn.adafruit.com/quickstart-adafruit-io-wippersnapper)
+- [Adafruit CircuitPython Installer](https://circuitpython.org/downloads)
+
+## Local development
+
+- Clone this repository.
+- Install dependencies with `npm`
```
Comment (dhalbert) [inline README.md:13]: There are no dependencies. Running `npm install` causes an error.
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+- Run `script/develop`
+- Open http://localhost:5004/
+
+## Origin
+
+This project was originally written by [Melissa LeBlanc-Williams](https://github.com/makermelissa). [Nabu Casa](https://www.nabucasa.com) ported the code over to TypeScript and in March 2022 took over maintenance from Adafruit. In July 2022, the Nabucasa stopped maintaining the project in favor of an official, but very early release of Espressif's [esptool-js](https://github.com/espressif/esptool-js/). Due to the instability of the tool, Adafruit updated their fork with Nabucasa's changes in November 2022 and took over maintenance once again.
```
Comment (dhalbert) [inline README.md:19]: This is out of date now.

**Review by dhalbert — CHANGES_REQUESTED** [review]
Some changes requested to the README.

I tested with a Feather ESP32-C6, a QT Py ESP32-C3, and a QT Py ESP32-S3.

C6:
```
Erasing flash (this may take a while)...
Error: Error: Timeout
```
But on the old one, it doesn't even get that far. So maybe there is still a problem with C6 support in the Espressif tools.

C3 and S3 work fine.

**Comment by tyeth** [comment]
Pyportal titano looks good
The fun house (s2) times out, but it's been fixed in latest esptool-js main
branch (pr160) so I've asked them to release it so unpkg gets updated.

**Comment by ladyada** [comment]
thanks, thats a good testbed

**Comment by tyeth** [comment]
Just a heads up that there is a minor delay on release for this C3 issue, but the fix went in today, so imagining a release tomorrow / wednesday: https://github.com/espressif/esptool-js/issues/161#issuecomment-2527316667

**Comment by tyeth** [comment]
@makermelissa v0.5.2 of esptool-js has been released.

**Comment by tyeth** [comment]
-Edit- It's got issues, atob fails when processing the stub, must be some bundling issue. When I run esptool-js's example and build to a site it works fine, but that doesn't use the cdn bundle (it uses the individually split files), I've filed a report: https://github.com/espressif/esptool-js/issues/167

**Comment by makermelissa** [comment]
Ok, I'll keep watching. It should point to the latest bundle automatically, but we can point it somewhere else besides the bundle if needed.

**Comment by makermelissa** [comment]
I have tied the esptool-js to version 0.5.1, which tested while working on the CircuitPython Installer so it won't break unexpectedly on us again.

File: README.md (hunk 2 above)
Comment (dhalbert) [reply README.md]: Update this history to say we went back to the Espressif version (or just remove the detailed history).

File: README.md (hunk 1 above)
Comment (dhalbert) [reply README.md]: Remove this line since there are no dependencies anymore.

**Review by dhalbert — CHANGES_REQUESTED** [review]
Just some README fixups requested.

**Comment by makermelissa** [comment]
> Just some README fixups requested.

Great suggestions. I'll update it.

**Review by dhalbert — APPROVED** [review]
Looks great!

**Pull request merged**